File: Arrays & Strings/001 2D Array/Search a 2D Matrix/Solution.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        for(int i= 0;i<matrix.length;i++) {
            int start = 0;
            int end = matrix[0].length-1;
            
            //Peform Binary search in each row
            while(start<=end) {
                int mid= start+ (end-start)/2;
                
                if(target== matrix[i][mid])
                    return true;
                else if (target< matrix[i][mid])
                    end=mid-1;
                else 
                    start = mid+1;
            }
        }
        
        return false;
    }
}

File: Arrays & Strings/001 2D Array/Search a 2D Matrix/Solution.kt
class Solution {
    fun searchMatrix(matrix: Array<IntArray>, target: Int): Boolean {
        for(i in 0..matrix.size-1) {
            var start = 0
            var end = matrix[0].size-1
            
            //Peform Binary search in each row
            while(start<=end) {
                var mid= start+ (end-start)/2
                
                when {
                    target== matrix[i][mid] -> return true
                    target< matrix[i][mid] -> end=mid-1
                    else -> start = mid+1
                }
            }
        }
        
        return false
    }
}
